### PR TITLE
Fix issues with HttpClientManager

### PR DESCRIPTION
**Pull request by Bond-009**
Fixes a crash when using the `Post` method with content that isn't a `byte` array.
Replaces #1460

### CODE REVIEW

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs [hunk 1]
```diff
@@ -23,30 +21,24 @@ namespace Emby.Server.Implementations.HttpClientManager
...
-
         private readonly IFileSystem _fileSystem;
         private readonly Func<string> _defaultUserAgentFn;
 
+        /// <summary>
+        /// Holds a dictionary of http clients by host.  Use GetHttpClient(host) to retrieve or create a client for web requests.
+        /// DON'T dispose it after use.
+        /// </summary>
+        /// <value>The HTTP clients.</value>
+        private readonly ConcurrentDictionary<string, HttpClient> _httpClients = new ConcurrentDictionary<string, HttpClient>();
```
Comment (cvium) [inline Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Is this necessary? Iirc, no state is even kept in httpclient

Comment (Bond-009) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: `HttpClient`s should be reused.
https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

Comment (cvium) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: My bad, I thought it was a custom http client.

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs [hunk 2]
```diff
@@ -109,45 +88,35 @@ private HttpRequestMessage GetRequestMessage(HttpRequestOptions options, HttpMet
...
-            {
-                request.Headers.Add(HeaderNames.Host, options.Host);
-            }
-
-            if (!string.IsNullOrEmpty(options.Referer))
-            {
-                request.Headers.Add(HeaderNames.Referer, options.Referer);
-            }
-
             //request.Headers.Add(HeaderNames.CacheControl, "no-cache");
```
Comment (cvium) [inline Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Remove?

**Pull request merged**